staging/src/k8s.io/apiserver/pkg/server/options/etcd.go

```diff
@@ -369,3 +359,56 @@ func WriteWatchCacheSizes(watchCacheSizes map[schema.GroupResource]int) ([]strin
 	}
 	return cacheSizes, nil
 }
+
+var _ serverstorage.StorageFactory = &StorageConfigFactory{}
+
+type StorageConfigFactory struct {
+	StorageConfig storagebackend.Config
+}
+
+func (s *StorageConfigFactory) NewConfig(resource schema.GroupResource) (*storagebackend.ConfigForResource, error) {
+	return s.StorageConfig.ForResource(resource), nil
+}
+
+func (s *StorageConfigFactory) ResourcePrefix(resource schema.GroupResource) string {
+	return resource.Group + "/" + resource.Resource
+}
+
+func (s *StorageConfigFactory) Backends() []serverstorage.Backend {
+	// nothing should ever call this method but we still provide a functional implementation
+	return (&serverstorage.DefaultStorageFactory{StorageConfig: s.StorageConfig}).Backends()
```

Comment:
slight preference for refactoring DefaultStorageFactory#Backends to:

```go
func (s *DefaultStorageFactory) Backends() []Backend {
	return backends(s.StorageConfig, s.Overrides)
}

func backends(storageConfig storagebackend.Config, storageOverrides map[schema.GroupResource]groupResourceOverrides) []Backend {
```

and making this call site `return backends(s.StorageConfig, nil)`